samples/system-test/products.v1p3beta1.test.js

```diff
@@ -57,32 +56,32 @@ async function getProductOrFalse(productPath) {
 
 describe(`products`, () => {
   before(tools.checkCredentials);
-
   before(async () => {
-    // Create a test product for each test
-    await productSearch.createProduct({
-      parent: productSearch.locationPath(
-        testProduct.projectId,
-        testProduct.location
-      ),
-      productId: testProduct.productId,
-      product: {
-        displayName: testProduct.productDisplayName,
-        productCategory: testProduct.productCategory,
-      },
-    });
-    testProduct.createdProductPaths.push(testProduct.productPath);
+    // Create a test product set for each test
+    try {
+      await productSearch.createProduct({
+        parent: productSearch.locationPath(
+          testProduct.projectId,
+          testProduct.location
+        ),
+        productId: testProduct.productId,
+        product: {
+          displayName: testProduct.productDisplayName,
+          productCategory: testProduct.productCategory,
+        },
+      });
+      testProduct.createdProductPaths.push(testProduct.productPath);
+    } catch (err) {} // ignore error
```

Comment:
Why is it safe to ignore the error here?  Is this required setup for the test? If it is safe to ignore, please add a comment in the code explaining why :)

Reply:
+1 – ignoring errors makes sense for deletions (may not have been created, tho I suppose we could check for this error explicitly) – if creation fails, it would be awesome if the `before()` section throws the error, rather than catching it incidentally from the body of the test

Reply:
Adding the `throw err;` back in `before()`

Reply:
Fixed.